### DISCUSSION TITLE
how to exclude some pre and post process time

### DISCUSSION THREAD

**Discussion by testwithproduction** [locustio/locust#2768] [Q&A] [answered]
assume we have a test that includes some readers and writers
writer will generate some random data, post to a rest api, and record a certain ID to a redis/db cache.
reader will read that id from the cache, and GET from a rest api.
certainly we do not want to include the time interacting with redis. wonder how to achieve that?
thanks

**Comment by testwithproduction** [comment]
looks like #243 tried to do something similar, is that still the only way? it is a 9y old post.

**Comment by cyberw** [comment]
Hi @testwithproduction ! There are some ways to exclude metrics, but in your case it probably makes sense to use your own http client (requests.Session instance) that doesnt log anything to locust, and instead fire request events manually.

That is done using self.environment.events.request.fire(...), see https://docs.locust.io/en/stable/testing-other-systems.html for more details.

If that is not clear enough, please post some of your code and I'll see if I can help you :)

**Comment by testwithproduction** [comment]
this is the idea
```import os
import redis
import uuid

from locust import FastHttpUser, task, between
from dotenv import load_dotenv

load_dotenv()  # take environment variables from .env

redis_url = os.getenv("REDIS_URL")

r = redis.Redis(redis_url)

class test(FastHttpUser):
    @task
    def get_stores(self, **kwargs):
        # does NOT want to be included in response time
        key = r.randomkey()
        data = r.get(key)
        # need to be included in response time        
        res = self.client.get("?id={0}".format(key), name="read", **kwargs)
        # does NOT want to be included in response time
        #post processing
    
    @task
    def post_stores_store_id_write(self, **kwargs):
        # does NOT want to be included in response time
        id = uuid.UUID()
        data = 'foo'

        # need to be included in response time
        self.client.post(
            "/write",
            name="write",
            json={'id':id, 'data':data}
            **kwargs
        )
        # does NOT want to be included in response time
        r.set(id, data)```

want to leverage the decorator. but want to exclude some expensive pre/post processing code.

**Comment by cyberw** [comment, marked as answer]
> want to leverage the decorator. but want to exclude some expensive pre/post processing code. 

Which decorator are you talking about? @task?

The code you posted should do exactly what you say you need, measure the response times of the http get/post, but not the redis calls. Tasks are never measured.

**Reply by testwithproduction** [reply]
I see, that is cool. thanks for clarifying
